base_layer/wallet/src/transaction_service/protocols/transaction_coinbase_monitoring_protocol.rs

```diff
@@ -154,275 +150,443 @@ where TBackend: TransactionBackend + 'static
-                            ));
-                        }
-
-                        break;
-                    }
-                }
-            }
-
-            if chain_metadata_response_received.is_some() && fetch_utxo_response_received {
-                debug!(
+                info!(
                     target: LOG_TARGET,
-                    "Both Base node responses received. TxId: {:?} not mined yet.", completed_tx.tx_id,
+                    "Coinbase monitoring process timed out for transaction (TxId: {})", self.tx_id
                 );
-                // Finish out the rest of this period before moving onto next round
-                delay.await;
-            }
-
-            info!(
-                target: LOG_TARGET,
-                "Coinbase monitoring process timed out for Transaction TX_ID: {}", completed_tx.tx_id
-            );
 
-            let _ = self
-                .resources
-                .event_publisher
-                .send(Arc::new(TransactionEvent::TransactionMinedRequestTimedOut(
-                    completed_tx.tx_id,
-                )))
-                .map_err(|e| {
-                    trace!(
-                        target: LOG_TARGET,
-                        "Error sending event, usually because there are no subscribers: {:?}",
+                let _ = self
+                    .resources
+                    .event_publisher
+                    .send(Arc::new(TransactionEvent::TransactionMinedRequestTimedOut(self.tx_id)))
+                    .map_err(|e| {
+                        trace!(
+                            target: LOG_TARGET,
+                            "Error sending event, usually because there are no subscribers: {:?}",
+                            e
+                        );
                         e
-                    );
-                    e
-                });
+                    });
+            }
         }
     }
 
-    async fn handle_base_node_response(
+    /// Attempt to query the location of the transaction from the base node via RPC.
+    /// # Returns:
+    /// `Ok((true, Some(u64)))`  => Transaction was successfully mined and confirmed.
+    /// `Ok((false, Some(u64)))` => Either the transaction is mined but does not have the required number of
+    ///                             confirmations yet, or it is not mined and still in the mempool, or it is not mined
+    ///                             and not found in the mempool.
+    /// `Ok((false, None))`      => There was a problem with the RPC call.
+    async fn query_coinbase_transaction(
         &mut self,
-        tx_id: TxId,
-        response: proto::BaseNodeServiceResponse,
-    ) -> Result<BaseNodeResponseType, TransactionServiceProtocolError>
+        signature: Signature,
+        completed_tx: CompletedTransaction,
+        client: &mut BaseNodeWalletRpcClient,
+    ) -> Result<(bool, Option<u64>), TransactionServiceProtocolError>
```

Comment:
Referring to my Style Nit above. If that logic lived in this function then this over complicated return type is not needed.

Reply:
Looked at the logic; I feel it would overcomplicate `async fn query_coinbase_transaction` if the `if tip > self.block_height + self.resources.config.num_confirmations_required {` logic is moved in there as well, as that is what would be needed to change the return type to 
``` rust
Result<(bool, bool), TransactionServiceProtocolError>
```